src/cryptography/convert.js

```diff
@@ -24,7 +24,18 @@ export const bufferToBigNumberString = bigNumberBuffer =>
 
 export const bufferToHex = buffer => naclInstance.to_hex(buffer);
 
-export const hexToBuffer = hex => Buffer.from(hex, 'hex');
+const hexRegex = /^[0-9a-f]+/i;
+export const hexToBuffer = hex => {
+	if (typeof hex !== 'string') {
+		throw new TypeError('Argument must be a string.');
+	}
+	const matchedHex = (hex.match(hexRegex) || [])[0];
+	if (!matchedHex) {
+		return Buffer.alloc(0);
+	}
+	const evenLength = Math.floor(matchedHex.length / 2) * 2;
+	return Buffer.from(matchedHex.slice(0, evenLength), 'hex');
```

Comment:
Do you think it is better to return a sliced element here rather than throwing an error in case it is not even?

Reply:
Same applies to non-hex strings

Reply:
I think main reason not to throw an error for non-even input is because node 8 is not behaving like that.

Reply:
Also, for the non-hex string, node 8 ignores the non hex string, and create the buffer from whatever it can from the front.
```
> process.version
'v8.9.4'
> Buffer.from('abcde', 'hex');
<Buffer ab cd>
> Buffer.from('abcdzz123', 'hex');
<Buffer ab cd>
```

Reply:
But this is our opportunity to make the behaviour we actually want.

Reply:
https://github.com/nodejs/node/pull/12012#issuecomment-289107822
I think main thing is that parallel behavior with `base64` and coherent eror handlings of various things it can happen

Reply:
Actually the sentiment seems to be like "you can do validation on your side if you want" on the node side of things: https://github.com/nodejs/node/issues/8569

And in my opinion I would prefer to have an error thrown.